In [1]:
#Import neccessary libraries
import pandas as pd
import numpy as np
import datetime


This dataset was downloaded from Mavin Analytics website (URL: https://www.mavenanalytics.io/data-playground)

It contains a year's worth of sales from a pizza place. It has 4 different tables (order_details, orders, pizza_types, and pizzas

In [2]:
#Load the csv files and create dataframes from them

df_order_details = pd.read_csv('/content/order_details.csv')
df_order = pd.read_csv('/content/orders.csv')
df_pizza_types = pd.read_csv('/content/pizza_types.csv', encoding = 'unicode_escape')
df_pizzas= pd.read_csv('/content/pizzas.csv')

In [3]:
#A sneak peak at the order_details data
df_order_details.head()

order_details_id  order_id       pizza_id  quantity
0                 1         1     hawaiian_m         1
1                 2         2  classic_dlx_m         1
2                 3         2  five_cheese_l         1
3                 4         2    ital_supr_l         1
4                 5         2     mexicana_m         1

In [4]:
#Check info to see if there are any missing values
df_order_details.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48620 entries, 0 to 48619
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   order_details_id  48620 non-null  int64 
 1   order_id          48620 non-null  int64 
 2   pizza_id          48620 non-null  object
 3   quantity          48620 non-null  int64 
dtypes: int64(3), object(1)
memory usage: 1.5+ MB


In [5]:
#A sneak peak at the orders data
df_order.head()

order_id        date      time
0         1  2015-01-01  11:38:36
1         2  2015-01-01  11:57:40
2         3  2015-01-01  12:12:28
3         4  2015-01-01  12:16:31
4         5  2015-01-01  12:21:30

In [6]:
#Check info to see if there are any missing values
df_order.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21350 entries, 0 to 21349
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   order_id  21350 non-null  int64 
 1   date      21350 non-null  object
 2   time      21350 non-null  object
dtypes: int64(1), object(2)
memory usage: 500.5+ KB


In [7]:
#A sneak peak at the pizza data
df_pizza_types.head()

pizza_type_id                          name category  \
0       bbq_ckn    The Barbecue Chicken Pizza  Chicken   
1      cali_ckn  The California Chicken Pizza  Chicken   
2   ckn_alfredo     The Chicken Alfredo Pizza  Chicken   
3     ckn_pesto       The Chicken Pesto Pizza  Chicken   
4    southw_ckn   The Southwest Chicken Pizza  Chicken   

                                         ingredients  
0  Barbecued Chicken, Red Peppers, Green Peppers,...  
1  Chicken, Artichoke, Spinach, Garlic, Jalapeno ...  
2  Chicken, Red Onions, Red Peppers, Mushrooms, A...  
3  Chicken, Tomatoes, Red Peppers, Spinach, Garli...  
4  Chicken, Tomatoes, Red Peppers, Red Onions, Ja...

In [8]:
#Check info to see if there are any missing values
df_pizza_types.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   pizza_type_id  32 non-null     object
 1   name           32 non-null     object
 2   category       32 non-null     object
 3   ingredients    32 non-null     object
dtypes: object(4)
memory usage: 1.1+ KB


In [9]:
##A sneak peak at the pizza data
df_pizzas.head()

pizza_id pizza_type_id size  price
0   bbq_ckn_s       bbq_ckn    S  12.75
1   bbq_ckn_m       bbq_ckn    M  16.75
2   bbq_ckn_l       bbq_ckn    L  20.75
3  cali_ckn_s      cali_ckn    S  12.75
4  cali_ckn_m      cali_ckn    M  16.75

In [10]:
#Check info to see if there are any missing values
df_pizzas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 96 entries, 0 to 95
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   pizza_id       96 non-null     object 
 1   pizza_type_id  96 non-null     object 
 2   size           96 non-null     object 
 3   price          96 non-null     float64
dtypes: float64(1), object(3)
memory usage: 3.1+ KB


KPI_1: Analysis of sales according to weekdays, this will give an insight into the weekday with the highest customer turnout

In [11]:
#We will need details from order_details and order tables, we join both dataframes to get desired data for analysis
df_orders_and_details = pd.merge(df_order_details, df_order, on='order_id', how='outer')

In [12]:
#Take a look at the resulting dataframe
df_orders_and_details.head()

order_details_id  order_id       pizza_id  quantity        date      time
0                 1         1     hawaiian_m         1  2015-01-01  11:38:36
1                 2         2  classic_dlx_m         1  2015-01-01  11:57:40
2                 3         2  five_cheese_l         1  2015-01-01  11:57:40
3                 4         2    ital_supr_l         1  2015-01-01  11:57:40
4                 5         2     mexicana_m         1  2015-01-01  11:57:40

In [13]:
#Extract day of the week from the date column
# Convert the 'date' column to datetime format
df_orders_and_details['date'] = pd.to_datetime(df_orders_and_details['date'])

# Extract the weekday and save it in a new column called 'Weekday'
df_orders_and_details['Weekday'] = df_orders_and_details['date'].dt.day_name()


In [14]:
#Resulting dataframe with the weekday column
df_orders_and_details.head()

order_details_id  order_id       pizza_id  quantity       date      time  \
0                 1         1     hawaiian_m         1 2015-01-01  11:38:36   
1                 2         2  classic_dlx_m         1 2015-01-01  11:57:40   
2                 3         2  five_cheese_l         1 2015-01-01  11:57:40   
3                 4         2    ital_supr_l         1 2015-01-01  11:57:40   
4                 5         2     mexicana_m         1 2015-01-01  11:57:40   

    Weekday  
0  Thursday  
1  Thursday  
2  Thursday  
3  Thursday  
4  Thursday

In [15]:
#To get the total sales for each weekday, group the ordres_and_details dataframe by weekday and sum the quantity for each day
#save to a new dataframe which will later be used for visualization purpose
df_sales_by_day = df_orders_and_details.groupby('Weekday')['quantity'].sum().reset_index()

In [16]:
#dataframe containing the results of the sales by weekday analysis
df_sales_by_day

Weekday  quantity
0     Friday      8242
1     Monday      6485
2   Saturday      7493
3     Sunday      6035
4   Thursday      7478
5    Tuesday      6895
6  Wednesday      6946

In [17]:
#Sorting by Quantity
df_sales_by_day = df_sales_by_day.sort_values(by=['quantity'])

In [18]:
#Save the resulting dataframe to a csv file called 'sales_by_weekday.csv'
df_sales_by_day.to_csv('sales_by_weekday.csv')

KPI_2: Analysis of peak hours

In [19]:
#Extract the hour from the time column and save in a new column
df_orders_and_details['hour'] = df_orders_and_details['time'].str[:2]

In [20]:
#Resulting dataframe from the operation performed in the previous code block
df_orders_and_details.head()

order_details_id  order_id       pizza_id  quantity       date      time  \
0                 1         1     hawaiian_m         1 2015-01-01  11:38:36   
1                 2         2  classic_dlx_m         1 2015-01-01  11:57:40   
2                 3         2  five_cheese_l         1 2015-01-01  11:57:40   
3                 4         2    ital_supr_l         1 2015-01-01  11:57:40   
4                 5         2     mexicana_m         1 2015-01-01  11:57:40   

    Weekday hour  
0  Thursday   11  
1  Thursday   11  
2  Thursday   11  
3  Thursday   11  
4  Thursday   11

In [21]:
#How many unique hours are there in the dataset? This will tell us how many hours of operation in a day
df_orders_and_details['hour'].unique()

array(['11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21',
       '22', '23', '10', '09'], dtype=object)

In [22]:
#The previous code block shows that the business is open from around 9am till 11pm

In [23]:
# To get the peak hours, group the orders_and_details dataframe by the hour and sum of quantitu for each hour and save in a new dataframe
df_sales_by_hours = df_orders_and_details.groupby('hour')['quantity'].sum().reset_index()

In [80]:
#Check the resulting dataframe
df_sales_by_hours

hour  quantity
3    12      6776
4    13      6413
9    18      5417
8    17      5211
10   19      4406
7    16      4239
5    14      3613
11   20      3534
6    15      3216
2    11      2728
12   21      2545
13   22      1386
14   23        68
1    10        18
0    09         4

In [78]:
#Sorting by Quantity to see the hour when the store is most busy
df_sales_by_hours = df_sales_by_hours.sort_values(by='quantity', ascending=False)

In [79]:
#Check to see the most busy hour of the day
most_busy_hour = df_sales_by_hours.head(1)
most_busy_hour

hour  quantity
3   12      6776

In [27]:
df_sales_by_hours.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 15 entries, 0 to 3
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   hour      15 non-null     object
 1   quantity  15 non-null     int64 
dtypes: int64(1), object(1)
memory usage: 360.0+ bytes


In [28]:
#Save to a new csv file for visualization purposes
df_sales_by_hours.to_csv('sales_by_hours.csv')

KPI_3: How many Pizzas are in a Typical order?

In [29]:
#This is calculated by getting the average of the quantity column of the orders_and_details_dataset

In [30]:
df_orders_and_details.loc[:, 'quantity'].mean()

1.0196215549156726

In [31]:
#What are the unique quantity values in the dataset?
df_orders_and_details['quantity'].unique()

array([1, 2, 3, 4])

In [32]:
#Check the quantity values and how many times they appear in the data
df_typical_order = df_orders_and_details['quantity'].value_counts()

In [33]:
df_typical_order

1    47693
2      903
3       21
4        3
Name: quantity, dtype: int64

In [34]:
#Save the information from the previous codeblock to a csv file for visualization analysis later
df_typical_order.to_csv('Nun_pizza_by_order.csv')

KPI_4: Are there any bestseller Pizzas?

In [35]:
#To get the needed information, we combine the information from the pizza_types and Pizza
df_interim = pd.merge(df_pizza_types, df_pizzas, on='pizza_type_id', how='outer')

In [36]:
#Join the resulting dataframe from previous code block to the orders_and_details dataframe
df_all = pd.merge(df_interim, df_orders_and_details, on='pizza_id', how='outer')

In [37]:
#View resulting dataframe
df_all.head()

pizza_type_id                        name category  \
0       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
1       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
2       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
3       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
4       bbq_ckn  The Barbecue Chicken Pizza  Chicken   

                                         ingredients   pizza_id size  price  \
0  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
1  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
2  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
3  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
4  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   

   order_details_id  order_id  quantity       date      time   Weekday hour  
0              11.0       6.0       1.0 2015-01-01  12:29:36  Thursday   12  
1             239.0     102.0       1.0 2015-01-02  17:54:04    Friday   17  
2             294.0     124.0       1.0 2015-01-02  20:12:34    Friday   20  
3             458.0     194.0       1.0 2015-01-03  21:21:24  Saturday   21  
4             489.0     208.0       1.0 2015-01-04  12:12:05    Sunday   12

In [38]:
df_all.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 48625 entries, 0 to 48624
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype         
---  ------            --------------  -----         
 0   pizza_type_id     48625 non-null  object        
 1   name              48625 non-null  object        
 2   category          48625 non-null  object        
 3   ingredients       48625 non-null  object        
 4   pizza_id          48625 non-null  object        
 5   size              48625 non-null  object        
 6   price             48625 non-null  float64       
 7   order_details_id  48620 non-null  float64       
 8   order_id          48620 non-null  float64       
 9   quantity          48620 non-null  float64       
 10  date              48620 non-null  datetime64[ns]
 11  time              48620 non-null  object        
 12  Weekday           48620 non-null  object        
 13  hour              48620 non-null  object        
dtypes: datetime64[ns](1), 

In [39]:
#Group the dataframe by the pizza name and sum of quantity for each name to tell us the ones with the most sales, save the result into a new dataframe
df_bestseller_pizza = df_all.groupby('name')['quantity'].sum().reset_index()

In [40]:
#A quick look at the first 5 rows of the dataframe
df_bestseller_pizza.head()

name  quantity
0    The Barbecue Chicken Pizza    2432.0
1            The Big Meat Pizza    1914.0
2          The Brie Carre Pizza     490.0
3           The Calabrese Pizza     937.0
4  The California Chicken Pizza    2370.0

In [41]:
len(df_bestseller_pizza)

32

In [42]:
# To get the bestsellers, Sort dataframe by 'quantity' column in descending order
df_bestseller_pizza = df_bestseller_pizza.sort_values(by='quantity', ascending=False)

In [43]:
#Pick the first 10 elements to get the first 10 bestselling pizzas
df_best_ten = df_bestseller_pizza.head(10)

In [44]:
df_best_ten

name  quantity
7       The Classic Deluxe Pizza    2453.0
0     The Barbecue Chicken Pizza    2432.0
12            The Hawaiian Pizza    2422.0
20           The Pepperoni Pizza    2418.0
30        The Thai Chicken Pizza    2371.0
4   The California Chicken Pizza    2370.0
23            The Sicilian Pizza    1938.0
26       The Spicy Italian Pizza    1924.0
25   The Southwest Chicken Pizza    1917.0
1             The Big Meat Pizza    1914.0

In [45]:
#save information to a csv file for visualization purposes later on
df_best_ten.to_csv('best_ten.csv')

In [46]:
#Pick the last 10 elements to get the first 10 worstselling pizzas
df_last_ten = df_bestseller_pizza.tail(10)

In [47]:
#save information to a csv file for visualization purposes later on
df_last_ten.to_csv('last_ten.csv')

In [48]:
df_last_ten

name  quantity
11        The Green Garden Pizza     997.0
5      The Chicken Alfredo Pizza     987.0
15  The Italian Vegetables Pizza     981.0
6        The Chicken Pesto Pizza     973.0
27       The Spinach Pesto Pizza     970.0
24         The Soppressata Pizza     961.0
28     The Spinach Supreme Pizza     950.0
3            The Calabrese Pizza     937.0
16       The Mediterranean Pizza     934.0
2           The Brie Carre Pizza     490.0

KPI_5: Total Revenue generated for the year

In [49]:
#To get the total revenue generated for the year, a function to calculate revenue is used
# A function to calculate revenue
def get_revenue(row):
    return row['price'] * row['quantity']



In [50]:
# Apply the get_revenue function to each row of the dataframe
df_all['revenue'] = df_all.apply(get_revenue, axis=1)



In [51]:
# Total revenue generated will be calculated as the sum of the valuesin the revenue column
total_revenue_generated = df_all['revenue'].sum()



In [52]:
df_all.head()

pizza_type_id                        name category  \
0       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
1       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
2       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
3       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
4       bbq_ckn  The Barbecue Chicken Pizza  Chicken   

                                         ingredients   pizza_id size  price  \
0  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
1  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
2  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
3  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
4  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   

   order_details_id  order_id  quantity       date      time   Weekday hour  \
0              11.0       6.0       1.0 2015-01-01  12:29:36  Thursday   12   
1             239.0     102.0       1.0 2015-01-02  17:54:04    Friday   17   
2             294.0     124.0       1.0 2015-01-02  20:12:34    Friday   20   
3             458.0     194.0       1.0 2015-01-03  21:21:24  Saturday   21   
4             489.0     208.0       1.0 2015-01-04  12:12:05    Sunday   12   

   revenue  
0    12.75  
1    12.75  
2    12.75  
3    12.75  
4    12.75

In [53]:

print("Total revenue generated for the year is: $", total_revenue_generated)


Total revenue generated for the year is: $ 817860.05


KPI_6: Analysis of seasonal sales

In [54]:
#seasonal sales
def extract_season(date_str):
    # Check if the date value is missing
    if pd.isnull(date_str):
        return "Missing"
    
    # Parse the date string into a datetime object
    date_obj = datetime.datetime.strptime(str(date_str), "%Y-%m-%d %H:%M:%S")
    
    # Extract the month component
    month = date_obj.month
    
    # Determine the season based on the month of the year
    if month in [12, 1, 2]:
        return "Winter"
    elif month in [3, 4, 5]:
        return "Spring"
    elif month in [6, 7, 8]:
        return "Summer"
    else:
        return "Fall"

In [55]:
#Apply the extract season function to the dataframe to help us determine seasonal sales
df_all['season'] = df_all['date'].apply(extract_season)

In [56]:
df_all.head()

pizza_type_id                        name category  \
0       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
1       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
2       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
3       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
4       bbq_ckn  The Barbecue Chicken Pizza  Chicken   

                                         ingredients   pizza_id size  price  \
0  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
1  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
2  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
3  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
4  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   

   order_details_id  order_id  quantity       date      time   Weekday hour  \
0              11.0       6.0       1.0 2015-01-01  12:29:36  Thursday   12   
1             239.0     102.0       1.0 2015-01-02  17:54:04    Friday   17   
2             294.0     124.0       1.0 2015-01-02  20:12:34    Friday   20   
3             458.0     194.0       1.0 2015-01-03  21:21:24  Saturday   21   
4             489.0     208.0       1.0 2015-01-04  12:12:05    Sunday   12   

   revenue  season  
0    12.75  Winter  
1    12.75  Winter  
2    12.75  Winter  
3    12.75  Winter  
4    12.75  Winter

In [57]:
#Group the dataframe by the season and sum of quantity for that season and save it to a new dataframe
df_best_season = df_all.groupby('season')['quantity'].sum().reset_index()

In [58]:
#Take a look at the dataframe
df_best_season.head()

season  quantity
0     Fall   12039.0
1  Missing       0.0
2   Spring   12740.0
3   Summer   12667.0
4   Winter   12128.0

In [59]:
#Group the dataframe by the season and sum of revenue for that season and save it to a new dataframe
df_best_season = df_all.groupby('season')['revenue'].sum().reset_index()

In [81]:
df_best_season

season    revenue
0     Fall  198603.00
1  Missing       0.00
2   Spring  210536.65
3   Summer  209066.35
4   Winter  199654.05

In [60]:
#Save the result to a csv file for visualization purposes later
df_best_season.to_csv('seasonal_revenue.csv')

KPI_7: Analysis of pizza size and its effect on sales

In [61]:
#check the sizes present in the dataset
df_all['size'].unique()

array(['S', 'M', 'L', 'XL', 'XXL'], dtype=object)

In [62]:
#Check the counts for each size in the dataset
df_all['size'].value_counts()

L      18527
M      15387
S      14139
XL       544
XXL       28
Name: size, dtype: int64

In [63]:
##Group the dataframe by the size and sum of quantity for that season and save it to a new dataframe
df_sales_by_size = df_all.groupby('size')['quantity'].sum().reset_index()

In [64]:
df_sales_by_size

size  quantity
0    L   18956.0
1    M   15635.0
2    S   14403.0
3   XL     552.0
4  XXL      28.0

In [65]:
#save to a csv file
df_sales_by_size.to_csv('size.csv')

In [66]:
#Check daily sales
df_daily = df_all.groupby('date')['quantity'].sum().reset_index()

In [67]:
df_daily.head()

date  quantity
0 2015-01-01     162.0
1 2015-01-02     165.0
2 2015-01-03     158.0
3 2015-01-04     106.0
4 2015-01-05     125.0

In [68]:
# To get the best and worst day of sales, Sort dataframe by 'quantity' column in descending order
df_daily = df_daily.sort_values(by='quantity', ascending=False)

In [69]:
df_daily.head()

date  quantity
323 2015-11-26     266.0
324 2015-11-27     264.0
283 2015-10-15     262.0
184 2015-07-04     234.0
183 2015-07-03     213.0

In [70]:
#best day
best_day = df_daily.head(1)
best_day

date  quantity
323 2015-11-26     266.0

In [71]:
#Worst day
worst_day = df_daily.tail(1)
worst_day

date  quantity
80 2015-03-22      77.0

In [72]:
# convert the date column to a pandas datetime object
df_all['date'] = pd.to_datetime(df_all['date'])

# extract the quarter from the datetime object and save it to a new column
df_all['quarter'] = df_all['date'].dt.quarter



In [73]:
df_all.head()

pizza_type_id                        name category  \
0       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
1       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
2       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
3       bbq_ckn  The Barbecue Chicken Pizza  Chicken   
4       bbq_ckn  The Barbecue Chicken Pizza  Chicken   

                                         ingredients   pizza_id size  price  \
0  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
1  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
2  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
3  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   
4  Barbecued Chicken, Red Peppers, Green Peppers,...  bbq_ckn_s    S  12.75   

   order_details_id  order_id  quantity       date      time   Weekday hour  \
0              11.0       6.0       1.0 2015-01-01  12:29:36  Thursday   12   
1             239.0     102.0       1.0 2015-01-02  17:54:04    Friday   17   
2             294.0     124.0       1.0 2015-01-02  20:12:34    Friday   20   
3             458.0     194.0       1.0 2015-01-03  21:21:24  Saturday   21   
4             489.0     208.0       1.0 2015-01-04  12:12:05    Sunday   12   

   revenue  season  quarter  
0    12.75  Winter      1.0  
1    12.75  Winter      1.0  
2    12.75  Winter      1.0  
3    12.75  Winter      1.0  
4    12.75  Winter      1.0

In [74]:
#Group the dataframe by the quarter and sum of revenue for that quarter and save it to a new dataframe
df_quarterly_sales = df_all.groupby('quarter')['revenue'].sum().reset_index()

In [75]:
df_quarterly_sales.head()

quarter    revenue
0      1.0  205350.00
1      2.0  208369.75
2      3.0  205016.20
3      4.0  199124.10

In [76]:
#Save to a csv file
df_quarterly_sales.to_csv('quarterly_sales.csv')

In [77]:
#save the combined dataframe to a csv file
df_all.to_csv('Full_data.csv')

From the analysis above, we can deduce the following conclusions
1. Best weekday is Friday.
2. The store is open from 9am - 11pm daily and the busiest hour is 12pm.
3. Quantities in a typical order is either 1, 2, 3, or 4 and most 1 is the most ordered quantity.
4. The classic Deluxe Pizza is the Number 1 best selling pizza.
5. The Brie Carrie Pizza is the lowest selling pizza.
6. The Total revenue generated for the year is 817,860.05.
7.The Spring season had the most generated revenue ($210, 536.65) than other seasons.
8. There are four sizes (L, M, S, XL, XXL) of pizzas produced and the L size had more sales in quantity (18, 527).
9. The best sales day of the year was 2015-11-26 with 266 quantity sold and the worst day of the year was 2015-3-22 with 77 pizzas sold.
10. The second quarter generated more revenue (208, 369.75) slightly above other quarters.






Visualization will be published on my Tableau public profile:
https://public.tableau.com/app/profile/gloria.abuka